libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/NSURL+SFStringUtils.m

```diff
@@ -116,4 +116,14 @@ + (void)appendPathComponents:(NSArray*)pathComponents toMutableUrlString:(NSMuta
     }
 }
 
+- (NSURL *)slashTerminatedUrl {
+    NSMutableString *urlString = [self.absoluteString mutableCopy];
+
+    if (NSMaxRange([self.path rangeOfString:@"/" options:NSBackwardsSearch]) < self.path.length - 1) {
```

Comment:
I actually put that code in there! But I seem to remember it actually broke one of the unit test cases I had written for the existing functionality. Overall this snippet was a little confusing to me because we are checking the path property, but then modifying the absoluteString.

So I basically left it alone and backed away slowly :)